cmake/modules/dts.cmake

```diff
@@ -398,7 +398,7 @@ execute_process(
   )
 
 if(NOT "${ret}" STREQUAL "0")
-  message(FATAL_ERROR "dtc failed with return code: ${ret}")
+  message(FATAL_ERROR "dtc failed with return code: ${ret}\n${stderr}")
```

Comment:
better to remove all this complex code and instead simply do:
```
execute_process(
  COMMAND ${DTC}
   ...
   ${ZEPHYR_DTS}
  OUTPUT_QUIET # Discard stdout
  WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
  COMMAND_ERROR_IS_FATAL ANY
  )
```

and then remove the
```
if(NOT "${ret}" STREQUAL "0")
 ...
```
logic below.

See also this comment for details: https://github.com/zephyrproject-rtos/zephyr/pull/78250#discussion_r1756603897

Reply:
So with COMMAND_ERROR_IS_FATAL, cmake only prints the output on failure? Or will will always prints the output no matter what with your proposed change?

Reply:
It will print any output on standard error regardless of the error code.

Basically:
- COMMAND_ERROR_IS_FATAL: Will fail CMake if return code is different from zero. Flag has no relation / impact on standard out or standard error.
- No `ERROR_OUTPUT`, will print any message from standard error in the normal CMake output.
   Just like we are doing with `message(... "... ${stderr}")`.

See: https://cmake.org/cmake/help/latest/command/execute_process.html
> If no OUTPUT_* or ERROR_* options are given the output will be shared with the corresponding pipes of the CMake process itself.

Note: `OUTPUT_QUIET` discard only standard out, not standard error.